api/package.json

```diff
@@ -131,7 +131,7 @@
     "db:migrate": "knex --knexfile db/knexfile.js migrate:latest && node scripts/database/pg-boss-migration.js",
     "db:rollback:latest": "knex --knexfile db/knexfile.js migrate:down",
     "db:prepare": "npm run db:delete && npm run db:create && npm run db:migrate",
-    "db:seed": "knex --knexfile db/knexfile.js seed:run",
+    "db:seed": "NODE_OPTIONS=\"--max-old-space-size=380\" knex --knexfile db/knexfile.js seed:run",
```

Comment:
remarque : dans la [doc de node](https://nodejs.org/api/cli.html#--max-old-space-sizesize-in-mib), ils recommandent de laisser 25% de RAM dispo "pour les autres usages et éviter du swapping". Si je lis bien l'info, côté Scalingo les containers S font 256MB mais je ne sais pas si il s'agit de RAM ou d'espace disque ? 

question 1 : pourquoi est-ce qu'on a besoin de tuner cette valeur depuis peu ? Est-ce une évolution attendue avec les mises à jour des seeds ?

question 2 : comment détermine-t-on cette valeur ? Et est-ce pérènne ? Y'a-t-il une approche où on peut gérer cette info via une autre méthode (ex : variable d'env avec valeur par défaut) ?

Je nage en eaux troubles, n'hésitez pas à m'expliquer si je suis à côté de la plaque.

Reply:
Les conteneurs S ont bien 256M de RAM, le disque est beaucoup plus gros mais nous ne l'utilisons quasiment pas car il est volatile (il ne vit que le temps du conteneur).

C'est effectivement une bonne pratique de laisser de la ram dispo pour l'os. En principe il le fait tout seul mais sur les conteneurs petits et dense (avec peu de ram laissé dispo), fixer la valeur lui permet d'en prendre plus que ce que son calcul lui aurait permis.

La bonne pratique c'est de laisser 25% pour l'OS ou 1Go sur les grosses machines (sur une machine à 128Go de ram, tu ne vas pas laisser 32Go à l'os ;) )

Reply:
Si tu ne laisse pas de ram à l'os, il va te forcer à swap (utiliser le disque dur en mémoire tampon) et cela est délétère pour les perfs car le disque dur est bcp plus lent que la ram (même avec les ssd).

Reply:
@yannbertrand on a besoin de le faire depuis peu car le référentiel de contenu augmente en taille et sa récupération / stockage dans Redis a récemment passé la limite mémoire.
Ce n'est pas une solution pérenne et cela demande des moyens pour investiguer et implémenter une solution long terme.
En attendant, cela permet de débloquer les déploiements de Review App.

Reply:
Est-ce qu'on a besoin du référentiel pour exécuter les seeds ? 🤔

Reply:
Oui pour certaines seeds qui nécessite de créer un user avec un profil pré-positionné, etc ...

Reply:
OK bien compris. Peut être qu'on a pas besoin de tout le référentiel ceci dit 🤔 ou bien ce n'est pas la bonne approche